tests/test_enumeration_management.py

```diff
@@ -0,0 +1,133 @@
+"""Test the SMP Enumeration Management group."""
+
+from __future__ import annotations
+
+from typing import Any, Dict, Type, TypeVar
+
+import cbor2
+import pytest
+from pydantic import BaseModel
+
+from smp import enumeration_management as smpenum
+from smp import header as smphdr
+from smp import message as smpmsg
+from tests.helpers import make_assert_header
+
+T = TypeVar("T", bound=smpmsg._MessageBase)
+
+
+def _do_test(
+    msg: Type[T],
+    op: smphdr.OP,
+    command_id: smphdr.CommandId.EnumManagement,
+    data: Dict[str, Any],
+    nested_model: Type[BaseModel] | None = None,
+) -> T:
+    cbor = cbor2.dumps(data, canonical=True)
+    assert_header = make_assert_header(smphdr.GroupId.ENUM_MANAGEMENT, op, command_id, len(cbor))
+
+    def _assert_common(r: smpmsg._MessageBase) -> None:
+        assert_header(r)
+        for k, v in data.items():
+            if type(v) is tuple and nested_model is not None:
+                for v2 in v:
+                    assert v2 == nested_model(**v2).model_dump()
+            else:
+                assert v == getattr(r, k)
+        assert cbor == r.BYTES[8:]
+
+    r = msg(**data)
+
+    _assert_common(r)  # serialize
+    _assert_common(msg.loads(r.BYTES))  # deserialize
+
+    return r
+
+
+def test_GroupCountRequest() -> None:
+    _do_test(
+        smpenum.GroupCountRequest,
+        smphdr.OP.READ,
+        smphdr.CommandId.EnumManagement.GROUP_COUNT,
+        {},
+    )
+
+
+def test_GroupCountResponse() -> None:
+    r = _do_test(
+        smpenum.GroupCountResponse,
+        smphdr.OP.READ_RSP,
+        smphdr.CommandId.EnumManagement.GROUP_COUNT,
+        {"count": 2},
+    )
+    assert r.count == 2
+
+
+def test_ListOfGroupsRequest() -> None:
+    _do_test(
+        smpenum.ListOfGroupsRequest,
+        smphdr.OP.READ,
+        smphdr.CommandId.EnumManagement.LIST_OF_GROUPS,
+        {},
+    )
+
+
+def test_ListOfGroupsResponse() -> None:
+    r = _do_test(
+        smpenum.ListOfGroupsResponse,
+        smphdr.OP.READ_RSP,
+        smphdr.CommandId.EnumManagement.LIST_OF_GROUPS,
+        {"groups": (2, 5, 15)},
+    )
+    assert r.groups == (2, 5, 15)
+
+
+@pytest.mark.parametrize("index", [0, 1, None])
+def test_GroupIdRequest(index: int | None) -> None:
+    _do_test(
+        smpenum.GroupIdRequest,
+        smphdr.OP.READ,
+        smphdr.CommandId.EnumManagement.GROUP_ID,
+        {"index": index} if index is not None else {},
```

Comment:
This looks correct to me. The Zephyr doc says it can be omitted to return the first group's groupId, and omitted means empty map. TBH it's frustrating for that to even be allowed when index 0 is so clear.

Feels like we don't need to support any abstractions for it. Unless I'm missing something, users should only use the details request since it contains all information of the other 3.